apps/balancer-tools/src/contexts/SelectedPoolRound.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import { useParams, useRouter } from "next/navigation";
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import invariant from "tiny-invariant";
+
+import { Round } from "#/app/apr/(utils)/rounds";
+
+interface RoundContextProps {
+  selectedRound: string | undefined;
+  selectedPool: string | undefined;
+  selectedNetwork: string;
+  setSelectedPool: (poolId: string | undefined) => void;
+  setSelectedRound: (roundId: string | undefined) => void;
+  setNetworkId: (networkId: string) => void;
+}
+
+export const RoundContext = createContext<RoundContextProps>(
+  {} as RoundContextProps,
+);
+
+interface RoundContextProviderProps {
+  children: ReactNode;
+}
+
+export const SelectedPoolRoundContextProvider = ({
+  children,
+}: RoundContextProviderProps) => {
+  const router = useRouter();
+  const [selectedRound, setSelectedRound] = useState<string | undefined>();
+  const [selectedPool, setSelectedPool] = useState<string | undefined>();
+  const [selectedNetwork, setNetworkId] = useState<string>("1");
```

Comment:
we already have this defined in the params, what do we gain by adding this complication?